go/oasis-test-runner/scenario/e2e/halt_restore.go

```diff
@@ -141,9 +141,16 @@ func (sc *haltRestoreImpl) Run(childEnv *env.Env) error {
 		return fmt.Errorf("scenario/e2e/halt_restore: failed waiting for halt epoch: %w", err)
 	}
 
-	// Wait additional few seconds for genesis docs to be dumped.
-	// XXX: fixed in #2296 where we wait for nodes to actually shutdown.
-	time.Sleep(3 * time.Second)
+	// Wait for validators to exit so that genesis docs are dumped.
```

Comment:
thx for doing this part! :+1: 

just wondering, why didn't we need to set validators to be `Restartable`?

Reply:
I've made the validators restartable too now. The errors are ignored because the validators are shutdown/crash in the time between network runs, when there's nobody watching the error channel (and on the second network run, a new channel is created).